installutils/helminstall/types/interfaces.go

```diff
@@ -1,30 +1,45 @@
 package types
 
 import (
+	"os"
+
+	"github.com/spf13/afero"
 	"helm.sh/helm/v3/pkg/action"
 	"helm.sh/helm/v3/pkg/chart"
 	"helm.sh/helm/v3/pkg/cli"
 	"helm.sh/helm/v3/pkg/release"
+	"k8s.io/client-go/tools/clientcmd"
 )
 
 //go:generate mockgen -source ./interfaces.go -destination ./mocks/mock_interfaces.go
 
+type HelmClientForFileConfigFactory func(kubeConfig, kubeContext string) HelmClient
+
+type HelmClientForMemoryConfigFactory func(config clientcmd.ClientConfig) HelmClient
+
 // This interface implements the Helm CLI actions. The implementation relies on the Helm 3 libraries.
 type HelmClient interface {
 	// Prepare an installation object that can then be .Run() with a chart object
-	NewInstall(kubeConfig, kubeContext, namespace, releaseName string, dryRun bool) (HelmInstaller, *cli.EnvSettings, error)
+	NewInstall(namespace, releaseName string, dryRun bool) (HelmInstaller, *cli.EnvSettings, error)
 
 	// Prepare an un-installation object that can then be .Run() with a release name
-	NewUninstall(kubeConfig, kubeContext, namespace string) (HelmUninstaller, error)
+	NewUninstall(namespace string) (HelmUninstaller, error)
 
 	// List the already-existing releases in the given namespace
-	ReleaseList(kubeConfig, kubeContext, namespace string) (ReleaseListRunner, error)
+	ReleaseList(namespace string) (ReleaseListRunner, error)
 
 	// Returns the Helm chart archive located at the given URI (can be either an http(s) address or a file path)
 	DownloadChart(chartArchiveUri string) (*chart.Chart, error)
```

Comment:
probably worth noting that this method will still use the filesystem, and will fail if your filesystem is readonly (so worth checking whether we depend on this in the csr agent installer)